File: app/src/main/java/com/kieran/winnipegbusbackend/TripPlanner/classes/RideSegment.java
package com.kieran.winnipegbusbackend.TripPlanner.classes;

import com.kieran.winnipegbusbackend.Route;

import org.json.JSONObject;

public class RideSegment extends Segment {
    Route route;

    public RideSegment(JSONObject segment, String fromTitle, String toTitle) {
        super(segment, fromTitle, toTitle);

        route = new Route(segment);
    }

    @Override
    public String toString() {
        return route.getRouteName();
    }
}

File: app/src/main/java/com/kieran/winnipegbusbackend/TripPlanner/classes/RideSegment.kt
package com.kieran.winnipegbusbackend.TripPlanner.classes

import com.kieran.winnipegbusbackend.Route

import org.json.JSONObject

class RideSegment(tripParameters: TripParameters, segment: JSONObject) : Segment(tripParameters, segment) {
    internal val route: Route = Route(segment)

    override fun toString(): String {
        return "Ride " + route.routeName.toString() + " to " + to?.title
    }
}
